django/thunderstore/api/cyberstorm/tests/test_communities.py

```diff
@@ -0,0 +1,264 @@
+from typing import Dict, List, Union
+
+import pytest
+from django.db.models import Count, Sum
+from django.http import HttpRequest
+from django.urls import reverse
+from rest_framework.exceptions import ParseError
+from rest_framework.test import APIClient
+
+from thunderstore.api.cyberstorm.views.community import CommunitiesAPIView
+from thunderstore.cache.enums import CacheBustCondition
+from thunderstore.cache.tasks import invalidate_cache
+from thunderstore.community.consts import PackageListingReviewStatus
+from thunderstore.community.factories import CommunityFactory, PackageListingFactory
+from thunderstore.community.models.community import Community
+from thunderstore.community.models.community_site import CommunitySite
+
+PARAMS_FIXTURE = {
+    "page_size": 1,
+    "page": 1,
+    "ordering": "name",
+}
+
+
+@pytest.fixture(scope="module", autouse=True)
+def clear_pagination_cache():
+    invalidate_cache(CacheBustCondition.any_package_updated)
+
+
+@pytest.mark.django_db
+def test_CommunitiesAPIView_regular_usage(
+    client: APIClient, community_site: CommunitySite
+):
+    response = client.get(
+        "/api/cyberstorm/c/",
+        HTTP_HOST=community_site.site.domain,
+    )
+    c = Community.objects.annotate(
+        pkgs=Count("package_listings", distinct=True),
+        downloads=Sum("package_listings__package__versions__downloads", distinct=True),
+    ).get(
+        identifier=community_site.community.identifier,
+    )
+    resp_com = response.json()["communities"][0]
+    assert response.status_code == 200
+    assert c.name == resp_com["name"]
+    assert c.identifier == resp_com["namespace"]
+    assert c.downloads == resp_com["download_count"]
+    assert c.pkgs == resp_com["package_count"]
+    assert (c.icon.url if bool(c.icon) else None) == resp_com["image_source"]
```

Comment:
This doesn't test the actual implementation any more after changes.

Reply:
Fixed this one